reference-architecture/azure-ansible/bastion.json

```diff
@@ -0,0 +1,251 @@
+          "vmSize" : "[parameters('vmSize')]"
+        },
+        "osProfile" : {
+          "computerName" : "[parameters('vmName')]",
+          "adminUsername" : "[parameters('adminUsername')]",
+          "linuxConfiguration" : {
+            "disablePasswordAuthentication" : "true",
+            "ssh" : {
+              "publicKeys" : [
+                {
+                  "path" : "[variables('sshKeyPath')]",
+                  "keyData" : "[parameters('sshKeyData')]"
+                }
+              ]
+            }
+          }
+        },
+        "storageProfile" : {
+          "imageReference" : "[parameters('imageReference')]",
+          "osDisk" : {
+            "name" : "osdisk",
+            "vhd" : {
+              "uri" : "[concat('http://', parameters('sa'), '.blob.core.windows.net/openshiftvmachines/', parameters('vmName'), '.vhd')]"
+            },
+            "caching" : "ReadWrite",
+            "createOption" : "FromImage"
+          },
+          "dataDisks" : [
+            {
+              "caching" : "None",
+              "createOption" : "Empty",
+              "diskSizeGB" : "128",
+              "lun" : 0,
+              "name" : "datadisk1",
+              "vhd" : {
+                "uri" : "[concat('http://', parameters('sa'), '.blob.core.windows.net/openshiftvmachines/', parameters('vmName'), '_datadisk1.vhd')]"
+              }
+            }
+          ]
+        },
+        "networkProfile" : {
+          "networkInterfaces" : [
+            {
+              "id" : "[resourceId('Microsoft.Network/networkInterfaces', concat(parameters('vmName'), 'nic'))]"
+            }
+          ]
+        }
+      }
+    },
+    {
+      "type" : "Microsoft.Compute/virtualMachines/extensions",
+      "name" : "[concat(parameters('vmName'),'/installcustomscript')]",
+      "apiVersion" : "[variables('apiVersion')]",
+      "location" : "[resourceGroup().location]",
+      "tags" : {
+        "displayName" : "VirtualMachineCustomScriptExtension"
+      },
+      "dependsOn" : [
+        "[concat('Microsoft.Compute/virtualMachines/', parameters('vmName'))]"
+      ],
+      "properties" : {
+        "publisher" : "Microsoft.OSTCExtensions",
+        "type" : "CustomScriptForLinux",
+        "typeHandlerVersion" : "1.3",
+        "settings" : {
+          "fileUris" : [
+            "[concat(parameters('baseTemplateUrl'), 'bastion.sh')]"
+          ],
+          "commandToExecute" : "[ concat('sh bastion.sh ', resourceGroup().name, ' ', parameters('adminUsername'), ' ', parameters('adminPassword'), ' ', reference(concat(parameters('vmName'), 'pip')).dnsSettings.fqdn, ' ', padLeft(parameters('numberOfNodes'), 2, '0'), ' ', parameters('routerExtIP'),' ', parameters('RHNUserName'), ' ', parameters('RHNPassword'),' ', parameters('SubscriptionPoolId'), ' ',  variables('sQuote'),  parameters('sshPrivateData'), variables('sQuote'),  ' ', variables('sQuote'), parameters('sshKeyData'), variables('sQuote')  ) ]"
```

Comment:
You are correct that the prefixing your credential variables with **RHN** has no material effect in this case, however, the reasoning behind @cooktheryan's Agile based suggestion of standardizing the credential variables by using the prefix **RHSM** over **RHN**, will go a long way in insuring that future refactoring efforts will be less painful. I'm certain that @cooktheryan wasn't trying to single you out in this comment, the avoidance of creating technical debt right from the start is very relevant and needs all of our attention. Standardizing these common variables is something we, as a team, all need to work together on. We could tack this on to #117 or create a new issue, but either way, doing the work up front will pay dividends in the future. /cc @dav1x @markllama @scollier